rle/rleplus_reader.go

```diff
@@ -1,9 +1,57 @@
 package rlepluslazy
 
 import (
+	"math/bits"
+
 	"golang.org/x/xerrors"
 )
 
+type decodeInfo struct {
+	length byte // length of the run
+	i      byte // i+1 is number of repeats of above run lengths
+	n      byte // number of bits to read
+	varint bool // varint signifies that futher bits need to be processed as a varint
```

Comment:
So, _technically_ we don't need this last flag as we can say "length == 0" means "varint". But that's probably too tricky.

Reply:
yeah, but to gain anything we would have to get it all to fit in 16 bits, so not worth it right now.

Reply:
Well, my thought was this plus nibbles. But yeah, not worth it.